### PR TITLE
chore(deps): upgrading lib to angular version 12

### PR DESCRIPTION
**Pull request by sapnildessai**
[#769 ](https://github.com/ghiscoding/Angular-Slickgrid/issues/769) : Angular 12 support

### CODE REVIEW

**Comment by ghiscoding** [comment]
There seems to be lot of issues with the Jest unit tests, I'm not sure if it's because I didn't update Angular-Jest in a while but there's definitely something wrong here. 
```shell
Test Suites: 31 failed, 135 passed, 166 total
Tests:       738 failed, 3 skipped, 2248 passed, 2989 total
```

**Comment by sapnildessai** [comment]
This is strange. When I run npm test on my local system it works pretty fine. I will take a look

File: tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,6 @@
     "noUnusedParameters": false,
     "noUnusedLocals": false,
     "noImplicitReturns": true,
-    "emitDecoratorMetadata": true,
```
Comment (ghiscoding) [inline tsconfig.json:15]: this line got removed and might be causing the Jest unit tests to fail on CircleCI

image

**Comment by ghiscoding** [comment]
@sapnildessai can you put back that line that I've identified in the last comment and see if that helps.

**Comment by ghiscoding** [comment]
different error now, can you remove the `flatpickr` from these 2 Filters component since as it says it will always be true anyway. I'll soon rewrite them anyway to JavaScript native element instead of jQuery

image

**Comment by ghiscoding** [comment]
now we're in business, it all works now 😃 

Was that line removal `"emitDecoratorMetadata": true` produced by running `ng update`? 
What might be better is to move that flag into the [tsconfig.spec.json](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/test/tsconfig.spec.json), which is the one used by Jest, and still remove it from the main `tsconfig.json`

**Comment by sapnildessai** [comment]
Yay!! Yes it was added by ng update. Yes, let's remove it from main tsconfig and add it to spec json file and see.

**Comment by ghiscoding** [comment]
that doesn't seem to work, so let's put it back to when it passed and I'll look at fixing it in future PR (unless you want to troubleshoot it). Thanks a lot for the great contribution here 👍

**Comment by ghiscoding** [comment]
@sapnildessai hey so I don't think you replied to question, I would just like to confirm... have you also tried the Excel Export (which was that issue related to)?

**Comment by sapnildessai** [comment]
Hey. Sorry which issue are you taking about? I haven't tried excel export after the upgrade.

**Comment by ghiscoding** [comment]
the issue #769, he wrote this

> I have upgraded my application to angular12. Am using server side slick grid of 2.29.1 . But Slick grid is working fine except export to excel option..

**Comment by sapnildessai** [comment]
No I haven't tried. Let me forcefully upgrade demo project and try it locally.

**Comment by sapnildessai** [comment]
I have verified that excel export is working fine after upgrading to angular 12. I did a force update on the demo fork.

**Pull request merged**